src/main/java/blackjack/domain/card/BlackjackScore.java

```diff
@@ -0,0 +1,107 @@
+package blackjack.domain.card;
+
+public record BlackjackScore(int value, int cardSize) {
+    private static final int BLACKJACK_CARD_NUMBER = 2;
+    private static final int ADDITIONAL_BLACKJACK_SCORE = 10;
+    private static final int DEALER_BLACKJACK_SCORE_THRESHOLD = 17;
+    private static final int BLACKJACK_SCORE = 21;
+    private static final BlackjackScore BLACKJACK_BLACKJACK_SCORE = new BlackjackScore(21, 2);
+
+    public BlackjackScore withAce() {
+        BlackjackScore maxBlackjackScore = this.addBlackjackCard();
+        if (maxBlackjackScore.isGreaterThan(BLACKJACK_SCORE)) {
+            return this;
+        }
+        return maxBlackjackScore;
+    }
+
+    private BlackjackScore addBlackjackCard() {
+        return new BlackjackScore(this.value + ADDITIONAL_BLACKJACK_SCORE, this.cardSize);
+    }
+
+    public WinningResult decide(BlackjackScore subBlackjackScore) {
+        if (isBlackjackWinning(subBlackjackScore)) {
+            return WinningResult.BLACKJACK_WIN;
+        }
+        if (isWinning(subBlackjackScore)) {
+            return WinningResult.WIN;
+        }
+        if (isLosing(subBlackjackScore)) {
+            return WinningResult.LOSE;
+        }
+        return WinningResult.DRAW;
+    }
+
+    public boolean isBlackjackWinning(BlackjackScore subBlackjackScore) {
+        if (this.isBlackjack() && !subBlackjackScore.isBlackjack()) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isWinning(BlackjackScore subBlackjackScore) {
+        if (!this.isBust() && subBlackjackScore.isBust()) {
+            return true;
+        }
+        if (!this.isBust()) {
+            return this.isGreaterThan(subBlackjackScore.value);
+        }
+        return false;
```

Comment:
```suggestion
        if (isBust()) {
            return false;
        }
        return subBlackjackScore.isBust() || isGreaterThan(subBlackjackScore.value);
```
bust면 무조건 지는 것이니 먼저 false를 내려보내고 나머지는 조건절 탈 필요없이 처리가 가능하지 않을까요?